### PR TITLE
HBASE-22729 Start RSGroupInfoManager as default

### PR DESCRIPTION
**Pull request by infraio**

### CODE REVIEW

File: hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupBasedLoadBalancer.java [hunk 1]
```diff
@@ -374,17 +370,12 @@ private void generateGroupMaps(
...
       for (RegionInfo region : regions) {
         RSGroupInfo targetRSGInfo = null;
-        try {
-          String groupName = rsGroupInfoManager.getRSGroupOfTable(region.getTable());
-          if (groupName == null) {
-            LOG.info("Group not found for table " + region.getTable() + ", using default");
-            groupName = RSGroupInfo.DEFAULT_GROUP;
-          }
-          targetRSGInfo = rsGroupInfoManager.getRSGroup(groupName);
-        } catch (IOException exp) {
```
Comment (Apache9) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupBasedLoadBalancer.java]: Why this change?

Comment (infraio) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupBasedLoadBalancer.java]: Seems the base branch is not right...

Comment (infraio) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupBasedLoadBalancer.java]: oh. The reason is the method didn't throw IOException now. So remove the try...catch thing.

**Comment by infraio** [comment]
> I think starting RSGroupInfoManager by default should be done after HBASE-22695? As no

What is the whole plan?

**Comment by Apache9** [comment]
https://github.com/Apache9/hbase/tree/HBASE-22695

I'm trying to store the rs group info to table first. It will lead to a huge change.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 62 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 3 new or modified test files. |
||| _ HBASE-22514 Compile Tests _ |
| +1 | mvninstall | 312 | HBASE-22514 passed |
| +1 | compile | 71 | HBASE-22514 passed |
| +1 | checkstyle | 89 | HBASE-22514 passed |
| +1 | shadedjars | 364 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 43 | HBASE-22514 passed |
| 0 | spotbugs | 342 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 338 | HBASE-22514 passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 348 | the patch passed |
| +1 | compile | 71 | the patch passed |
| +1 | javac | 71 | the patch passed |
| +1 | checkstyle | 101 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 363 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 966 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 43 | the patch passed |
| +1 | findbugs | 346 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 14806 | hbase-server in the patch failed. |
| +1 | asflicense | 31 | The patch does not generate ASF License warnings. |
| | | 18746 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-405/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/405 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux eddc57e48779 4.4.0-154-generic #181-Ubuntu SMP Tue Jun 25 05:29:03 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-405/out/precommit/personality/provided.sh |
| git revision | HBASE-22514 / 7a0b82209c |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-405/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-405/1/testReport/ |
| Max. process+thread count | 4576 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-405/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by infraio** [comment]
> https://github.com/Apache9/hbase/tree/HBASE-22695
> 
> I'm trying to store the rs group info to table first. It will lead to a huge change.

Got it. Will continue this issue later.
